TACC/protx-dashboard
# Task/COOKS-345: fix redirecting when setupcomplete is false

## Overview: ##

This PR cleans up the redirecting when `setupcomplete=false` so that user is notified or routed to onboarding page.

This PR also fixes the login issue we were ignoring in our dev environments.

## Related Jira tickets: ##

* [COOKS-345](https://jira.tacc.utexas.edu/browse/COOKS-345)

## Testing Steps: ##
1. Follow README to set up snippet in the CMS for /analytics which has the iframe pointing to to /protx/dash/analytics
2. Login for the first time and see that your onboarding takes to you to /analytics.  (if you have the dev tools open you will see the new verify-user redirect
3. Logout
4. Remove your user (https://github.com/TACC/Core-Portal/wiki/How-to-Delete-Your-User)
5. Change your onboarding steps to be a wrong project membership (59024 instead of 57877 in `conf/portal/settings_custom.py`.  59024 corresponds to a test project that we aren't a member of https://tas.tacc.utexas.edu/TACCAdministration/Projects/EditProject.aspx?id=59024)
6. Restart containers (as the worker service doesn't automatically get these changes): `docker compose down` then ` docker compose up`
7.  Login and confirm that you are redirected to /workbench/analytics
8. Click on Analytics menu bar.  Confirm that you get a message with a link to /workbench/onboarding/ 

# Notes
- [x] add unit test for new routes?
- [x] front end unit test
